pkg/cmd/openshift-controller-manager/controller/image.go

```diff
@@ -63,51 +58,41 @@ func (c *ImageTriggerControllerConfig) RunController(ctx ControllerContext) (boo
 			Reactor:   &triggerdeploymentconfigs.DeploymentConfigReactor{Client: appsClient.Apps()},
 		},
 	}
-	if !c.HasBuilderEnabled {
-		sources = append(sources, imagetriggercontroller.TriggerSource{
-			Resource:  schema.GroupResource{Group: "build.openshift.io", Resource: "buildconfigs"},
-			Informer:  ctx.BuildInformers.Build().InternalVersion().BuildConfigs().Informer(),
-			Store:     ctx.BuildInformers.Build().InternalVersion().BuildConfigs().Informer().GetIndexer(),
-			TriggerFn: triggerbuildconfigs.NewBuildConfigTriggerIndexer,
-			Reactor:   triggerbuildconfigs.NewBuildConfigReactor(bcInstantiator, kclient.Core().RESTClient()),
-		})
-	}
-	if !c.HasDeploymentsEnabled {
-		sources = append(sources, imagetriggercontroller.TriggerSource{
-			Resource:  schema.GroupResource{Group: "extensions", Resource: "deployments"},
-			Informer:  ctx.ExternalKubeInformers.Extensions().V1beta1().Deployments().Informer(),
-			Store:     ctx.ExternalKubeInformers.Extensions().V1beta1().Deployments().Informer().GetIndexer(),
-			TriggerFn: triggerannotations.NewAnnotationTriggerIndexer,
-			Reactor:   &triggerannotations.AnnotationReactor{Updater: updater},
-		})
-	}
-	if !c.HasDaemonSetsEnabled {
-		sources = append(sources, imagetriggercontroller.TriggerSource{
-			Resource:  schema.GroupResource{Group: "extensions", Resource: "daemonsets"},
-			Informer:  ctx.ExternalKubeInformers.Extensions().V1beta1().DaemonSets().Informer(),
-			Store:     ctx.ExternalKubeInformers.Extensions().V1beta1().DaemonSets().Informer().GetIndexer(),
-			TriggerFn: triggerannotations.NewAnnotationTriggerIndexer,
-			Reactor:   &triggerannotations.AnnotationReactor{Updater: updater},
-		})
-	}
-	if !c.HasStatefulSetsEnabled {
-		sources = append(sources, imagetriggercontroller.TriggerSource{
-			Resource:  schema.GroupResource{Group: "apps", Resource: "statefulsets"},
-			Informer:  ctx.ExternalKubeInformers.Apps().V1beta1().StatefulSets().Informer(),
-			Store:     ctx.ExternalKubeInformers.Apps().V1beta1().StatefulSets().Informer().GetIndexer(),
-			TriggerFn: triggerannotations.NewAnnotationTriggerIndexer,
-			Reactor:   &triggerannotations.AnnotationReactor{Updater: updater},
-		})
-	}
-	if !c.HasCronJobsEnabled {
-		sources = append(sources, imagetriggercontroller.TriggerSource{
-			Resource:  schema.GroupResource{Group: "batch", Resource: "cronjobs"},
-			Informer:  ctx.ExternalKubeInformers.Batch().V2alpha1().CronJobs().Informer(),
-			Store:     ctx.ExternalKubeInformers.Batch().V2alpha1().CronJobs().Informer().GetIndexer(),
-			TriggerFn: triggerannotations.NewAnnotationTriggerIndexer,
-			Reactor:   &triggerannotations.AnnotationReactor{Updater: updater},
-		})
-	}
+	sources = append(sources, imagetriggercontroller.TriggerSource{
```

Comment:
I'm slightly worried we're enabling this, iirc this was explicitly disabled due to some problems. @tnozicka or @mfojtik mind remember some details.

Reply:
> I'm slightly worried we're enabling this, iirc this was explicitly disabled due to some problems. @tnozicka or @mfojtik mind remember some details.

It would have failed validation here, right?  https://github.com/openshift/origin/pull/19070/files#diff-6b6e279f3c86e5edb0688c1ca449db56L241